docs/diagrams/wallet-lifecycle/wallet-state-transition.tikz

```diff
@@ -0,0 +1,138 @@
+                                                                               - notifyMovingFundsBelowDust}         (closing);
+\draw[->](closing)      --  node[below right, pos=0, yshift=-5mm, align=left] {- notifyWalletClosingPeriodElapsed}   (closed);
+\draw[->](live)         -|  node[above right, pos=0, xshift=5mm, align=left]  {- notifyWalletHeartbeatFailed *\\
+                                                                               - notifyCloseableWallet *\\
+                                                                               - notifyRedemptionTimeout *\\
+                                                       \sffamily\footnotesize{(* only if there's no main UTXO)}}     +(10cm,0)
+                        |-                                                                                           (closing);
+\draw[->](live)         -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout\\
+                                                                               - notifyMovedFundsSweepTimeout}       ([xshift=-50mm]terminated);
+\draw[->](movingFunds)  -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout\\
+                                                                               - notifyMovingFundsTimeout\\
+                                                                               - notifyMovedFundsSweepTimeout}       (terminated);
+\draw[->](closing)      -|  node[above left, pos=0, xshift=-5mm, align=right] {- notifyFraudChallengeDefeatTimeout}  ([xshift=15mm]terminated);
+
+
+% TEXT BOX
+
+\node [box, below = of closed, rectangle, xshift=-10mm, yshift=10mm] (box_label)
+{FUNCTIONS AVAILABLE PER STATE\\ \hfill};
+
+\node [box, below of=box_label, rectangle split, rectangle split horizontal, rectangle split parts=3, yshift=-5mm,](box_content)
+{
+LIVE:\\
+  \ttfamily{
+  - revealDeposit\\
+  - submitDepositSweepProof\\
+  - submitMovedFundsSweepProof\\
+  - notifyMovedFundsSweepTimeout\\
+  - requestRedemption\\
+  - submitRedemptionProof\\
+  - submitFraudChallenge\\
+  - defeatFraudChallenge\\
+  - notifyFraudChallengeDefeatTimeout\\
+  - notifyWalletHeartbeatFailed\\
+  - notifyCloseableWallet\\
+  - notifyRedemptionTimeout}
+\nodepart{two} 
+  MOVING\_FUNDS:\\
+  \ttfamily{
+  - submitDepositSweepProof\\
+  - submitMovedFundsSweepProof\\
+  - notifyMovedFundsSweepTimeout\\
+  - submitRedemptionProof\\
+  - submitFraudChallenge\\
+  - defeatFraudChallenge\\
+  - notifyFraudChallengeDefeatTimeout\\
+  - notifyRedemptionTimeout\\
+  - resetMovingFundsTimeout \\
+  - submitMovingFundsCommitment\\
+  - submitMovingFundsProof\\
+  - notifyMovingFundsTimeout\\
+  - notifyMovingFundsBelowDust}
+\nodepart{three}
+  CLOSING:\\
+  \ttfamily{
+  - submitFraudChallenge\\
+  - defeatFraudChallenge\\
+  - notifyFraudChallengeDefeatTimeout\\
+  - notifyWalletClosingPeriodElapsed}
```

Comment:
Let's put functions in the same order as in the `Bridge` contract and add the missing `defeatFraudChallengeWithHeartbeat`:

```suggestion
  - notifyWalletClosingPeriodElapsed\\
  - submitFraudChallenge\\
  - defeatFraudChallenge\\
  defeatFraudChallengeWithHeartbeat\\
  - notifyFraudChallengeDefeatTimeout}
```

Reply:
5809abf8b408aa7bb1a7478068ff1616c5270b89